apps/site/assets/ts/schedule/components/ScheduleFinder.tsx

```diff
@@ -82,16 +82,21 @@ const ScheduleFinder = ({
 
   React.useEffect(
     () => {
-      const newDirection = query.direction_id === "0" ? 0 : 1;
+      const scheduleFinder = {
```

Comment:
@alicia-mbta so this is not quite what I meant. The concern here is not what the variable names are going to be in the Typescript code, but rather what the keys of the query parameters are going to be. As it stands we are using the key `direction_id` for two separate things, one in the schedule finder, one in the direction pickers. This is going to lead to bugs if we let it stand.

My solution is to change the names of these params in the query string, so that they're called "schedule_finder[direction_id]" and "schedule_direction[direction_id]" respectively. For the sake of consistency, we should also change the name of the `origin` param here to `schedule_finder[origin]`, and the name of the `variant` param in the direction picker to `schedule_direction[variant]`.

When you change the names of the query params in the frontend, of course, you'll need to change them in the backend to match. Because of the way Elixir parses query params, if you use the names I suggest, you'll end up with nested query params on the Elixir side, which is what we want in this case.

So what I suggest is that you revert this commit and try changing the query parameter names as I describe. If you want we could pair later in the day on this, in case what I'm saying is not clear.

Reply:
Made some more changes, please review them when you can because I'm still unsure of whether I'm still missing something or not.
Also investigating why `npm run format:check` is failing.

Reply:
one thing about formatting, is you can also do `npm run format` and that'll automatically format your JS/TS code to match the standard.